### DISCUSSION TITLE
I am using the API to download scheduled

### DISCUSSION THREAD

**Discussion by Siva700** [CrowdStrike/falconpy#861] [Q&A] [answered]
Is there anyway i can download scheduled repot please using falconpy module ?

**Comment by crowdstrikedcs** [comment, marked as answer]
Hi @Siva700 thanks for the question!

We do provide a Service Collection for the Scheduled Reports API [here](https://falconpy.io/Service-Collections/Scheduled-Reports.html)

The one caveat being that scheduled reports are currently only able to be created from the Falcon UI. 

Generally, after the creation of a scheduled report, we can use FalconPy to:

1. [Query for scheduled reports ](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_query) a proper query will return a report ID that we will use later for getting report details.

2. [Launch (execute) a scheduled report.](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_launch) If a report has not yet run on it's schedule we can run it manually. This also should meet the need of ad-hoc execution.

3. [Get the report details](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_get) allowing you to receive output from the report

We have some coding examples included with the documentation above, as well as a full sample [here](https://github.com/CrowdStrike/falconpy/blob/main/samples/report_executions/get_report_results.py) take a look and let us know with any questions!

**Reply by Siva700** [reply]
Thank you. My schedule report runs weekly. I want to download the latest report from the report history as a csv format. Is it anything you can advice please.

**Reply by Siva700** [reply]
All sorted now. Those are great python module. Thank you

**Reply by crowdstrikedcs** [reply]
When creating a report, you have the option to set the results file format to either CSV, JSON, or PDF. 
![image](https://user-images.githubusercontent.com/43282232/208706006-4df73e63-fa96-49f6-85fb-4eb308f53403.png)

To get the latest execution for a report, you should grab the ```scheduled_report_id``` from the Falcon UI and use it as a ```filter``` in FalconPy like this:

```response = falcon.query_reports(sort="last_execution_on",filter="scheduled_report_id:'XXX'")```

Given the ```scheduled_report_id``` this will return execution IDs sorted by execution date
